Source/Core/Core/IOS/IPC.cpp

```diff
@@ -85,6 +86,302 @@ static u64 s_last_reply_time;
 
 static constexpr u64 ENQUEUE_REQUEST_FLAG = 0x100000000ULL;
 static constexpr u64 ENQUEUE_ACKNOWLEDGEMENT_FLAG = 0x200000000ULL;
+
+struct IosMemoryValues
+{
+  u16 ios_number;
+  u32 ios_version;
+  u32 ios_date;
+  u32 mem1_physical_size;
+  u32 mem1_simulated_size;
+  u32 mem1_end;
+  u32 mem1_arena_begin;
+  u32 mem1_arena_end;
+  u32 mem2_physical_size;
+  u32 mem2_simulated_size;
+  u32 mem2_end;
+  u32 mem2_arena_begin;
+  u32 mem2_arena_end;
+  u32 ipc_buffer_begin;
+  u32 ipc_buffer_end;
+  u32 hollywood_revision;
+  u32 ram_vendor;
+  u32 unknown_begin;
+  u32 unknown_end;
+};
+
+constexpr u32 ADDR_MEM1_SIZE = 0x3100;
+constexpr u32 ADDR_MEM1_SIM_SIZE = 0x3104;
+constexpr u32 ADDR_MEM1_END = 0x3108;
+constexpr u32 ADDR_MEM1_ARENA_BEGIN = 0x310c;
+constexpr u32 ADDR_MEM1_ARENA_END = 0x3110;
+constexpr u32 ADDR_PH1 = 0x3114;
+constexpr u32 ADDR_MEM2_SIZE = 0x3118;
+constexpr u32 ADDR_MEM2_SIM_SIZE = 0x311c;
+constexpr u32 ADDR_MEM2_END = 0x3120;
+constexpr u32 ADDR_MEM2_ARENA_BEGIN = 0x3124;
+constexpr u32 ADDR_MEM2_ARENA_END = 0x3128;
+constexpr u32 ADDR_PH2 = 0x312c;
+constexpr u32 ADDR_IPC_BUFFER_BEGIN = 0x3130;
+constexpr u32 ADDR_IPC_BUFFER_END = 0x3134;
+constexpr u32 ADDR_HOLLYWOOD_REVISION = 0x3138;
+constexpr u32 ADDR_PH3 = 0x313c;
+constexpr u32 ADDR_IOS_VERSION = 0x3140;
+constexpr u32 ADDR_IOS_DATE = 0x3144;
+constexpr u32 ADDR_UNKNOWN_BEGIN = 0x3148;
+constexpr u32 ADDR_UNKNOWN_END = 0x314c;
+constexpr u32 ADDR_PH4 = 0x3150;
+constexpr u32 ADDR_PH5 = 0x3154;
+constexpr u32 ADDR_RAM_VENDOR = 0x3158;
+constexpr u32 ADDR_BOOT_FLAG = 0x315c;
+constexpr u32 ADDR_APPLOADER_FLAG = 0x315d;
+constexpr u32 ADDR_DEVKIT_BOOT_PROGRAM_VERSION = 0x315e;
+constexpr u32 ADDR_SYSMENU_SYNC = 0x3160;
+
+constexpr u32 MEM1_SIZE = 0x01800000;
+constexpr u32 MEM1_END = 0x81800000;
+constexpr u32 MEM1_ARENA_BEGIN = 0x00000000;
+constexpr u32 MEM1_ARENA_END = 0x81800000;
+constexpr u32 MEM2_SIZE = 0x4000000;
+constexpr u32 MEM2_ARENA_BEGIN = 0x90000800;
+constexpr u32 HOLLYWOOD_REVISION = 0x00000011;
+constexpr u32 PLACEHOLDER = 0xDEADBEEF;
+constexpr u32 RAM_VENDOR = 0x0000FF01;
+constexpr u32 RAM_VENDOR_MIOS = 0xCAFEBABE;
+
+constexpr std::array<IosMemoryValues, 31> ios_memory_values = {
+    {{
```

Comment:
Would it make sense to disable clang-format for `ios_memory_values`? The formatting looks a bit weird and makes it hard to figure out what value is used for